opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -95,14 +95,141 @@ message InstrumentationLibraryMetrics {
+// and then re-initializes it.  StartTimeUnixNano is critical for
+// calculating correct rates for Monotonic Sum, Histogram, and Summary
+// data points in both of these cases.
+//
+// Another use for this information which applies to all point kinds
+// is to prevent overlap when multiple writers accidentally write to a
+// single stream.  StartTimeUnixNano MAY be used for overlap detection
+// for all point kinds.
+//
+// ## StartTimeUnixNano: Sums and Histograms
+//
+// Since these two point kinds have AggregationTemporality, the use of
+// StartTimeUnixNano is strongly recommended to ensure correct rate
+// calculation and prevent overlap.
+//
+// ## StartTimeUnixNano: Gauges
+//
+// The use of StartTimeUnixNano is optional for Gauge points.  This is
+// recommended in cases where there is a risk of mis-interpretion caused
+// by overlapping points (by a violation of the single-writer principle).
+//
+// When used, StartTimeUnixNano should be set to the start time of the
+// process or the timestamp when the Gauge was first set.
+//
+// ## StartTimeUnixNano: Summaries
+//
+// Summary points are treated the same as for cumulative aggregation
+// temporality, by convention.  StartTimeUnixNano should be set to the
+// starting timestamp in an unbroken sequence of observations.
+//
+// # Unknown StartTimeUnixNano
+//
+// This protocol adopts the Prometheus heuristic for reset detection,
+// and with it prescribes a solution for filling in StartTimeUnixNanos
+// with as much information as possible to assist consumers with
+// calculating rates and detecting overlap.  This process can be
+// applied to Sum and Histogram points with cumulative aggregation
+// temporality, Gauge, and Summary points.
+//
+// The Prometheus heuristic indicates that a reset must have occured
+// in a monotonic series whenever the value descends from one value to
+// the next, in a timestamp-ordered sequence.
+//
+// To facilitate reset and overlap detection, when single-timestamped
+// sample values are translated into OTLP, such as from the Prometheus
+// Remote-Write protocol, use the following.
+//
+// ## Filling in StartTimeUnixNano
+//
+// Note: This is meant to provide guidance in importing data into OTLP
+// that does not have start time and there is no other form of reset
+// detection.  This applies to points with cumulative aggregation
+// temporality and Gauge points but cannot be applied to points with
+// delta aggregation temporality.
+//
+// To import timestamped measurements into into OTLP using a stateful
+// processor, the input must be processed in order.  A series of
+// points that formed the beginning of the sequence, including an
+// unknown number of resets, are assumed to be lost or previously
+// recorded when the processor begins observing points.  Then:
+//
+// 1. Keep a map, using the metric name, resource, and attributes as
+//    the key, containing a struct of three parts:
```

Comment:
Probably instrumentation library as well? Maybe refer to specification "metric identity" ?

Reply:
I added instrumentation library. I don't know that the "identity" section is written, but I wonder how to think about instrumentation library version. I'd ignore it, but don't know how to explain why.